engine/src/main/java/org/kigalisim/engine/state/Scope.java

```diff
@@ -19,6 +19,13 @@
  * Object defining a scope within the engine including variables accessible from that scope.
  *
  * <p>Manages scope hierarchy and variable access across different context levels in the engine.</p>
+ * 
+ * <p>Key differences from JavaScript implementation:
```

Comment:
Thank you for this! However, I think I will document this elsewhere. Please remove this new paragraph (`<p>`)